lms/djangoapps/instructor/services.py

```diff
@@ -78,13 +76,60 @@ def delete_student_attempt(self, student_identifier, course_id, content_id, requ
             except (StudentModule.DoesNotExist, enrollment.sub_api.SubmissionError):
                 err_msg = (
                     'Error occurred while attempting to reset student attempts for user '
-                    '{student_identifier} for content_id {content_id}.'.format(
-                        student_identifier=student_identifier,
-                        content_id=content_id
-                    )
+                    f'{student_identifier} for content_id {content_id}.'
                 )
                 log.error(err_msg)
 
+    def complete_student_attempt(self, user_identifier, content_id):
+        """
+        """
+        try:
+            user = get_user_by_username_or_email(user_identifier)
+        except ObjectDoesNotExist:
+            err_msg = (
+                'Error occurred while attempting to complete student attempt for user '
+                f'{user_identifier} for content_id {content_id}. '
+                'User does not exist!'
+            )
+            log.error(err_msg)
+            return
+
+        try:
+            block_key = UsageKey.from_string(content_id)
+        except InvalidKeyError:
+            err_msg = (
+                'Error occurred while attempting to complete student attempts for user '
+                f'{user_identifier} for content_id {content_id}. '
+                'Invalid content_id!'
+            )
+            log.error(err_msg)
+            return
+
+        def _submit_completions(block, user):
+            """
```

Comment:
is this docstring unintentionally empty?

Reply:
Whoops, meant to specify this PR is a WIP/draft so I haven't done things like fill in docstrings/add tests yet. All incoming :)